keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -172,7 +173,7 @@ class ArtifactListener(
    */
   private fun debStatus(artifact: Artifact): ArtifactStatus {
     val status = artifact.metadata["releaseStatus"]?.toString()
-      ?: throw IllegalStateException("Artifact event received without 'releaseStatus' field")
+      ?: throw IntegrationException("Artifact event received without 'releaseStatus' field")
```

Comment:
I think here `IntegrationException` is less clear. I think it would be more clear if we had an `IllegalStateException` that extended the `IntegrationException`. Thoughts?

Reply:
Also seems like a system exception to me

Reply:
You know, I used `IntegrationException` here because of this as artifact events coming from echo or igor, but this is an internal event handler. I looked at the other available kork exceptions to see if there was something closer in meaning but didn't find anything. Would you be ok with just changing to `SystemException`?

Reply:
Actually, now that I read through your PR more, I think it's okay as it is. It is an `IntegrationException` because this is coming from echo.